genesis/src/lib.rs

```diff
@@ -69,23 +58,33 @@ impl GenesisNetwork {
         let transactions_iter = raw_block.transactions.into_iter();
 
         let transactions = transactions_iter
-            .map(|raw_transaction| {
-                raw_transaction.sign(genesis_key_pair.clone())
-            })
             .enumerate()
-            .filter_map(|(i, res)| {
-                res.map_err(|error| {
-                    let error_msg = format!("{error:#}");
-                    iroha_logger::error!(error = %error_msg, transaction_num=i, "Genesis transaction failed")
-                })
-                .ok()
-            }).map(GenesisTransaction)
-            .collect::<Vec<_>>();
+            .map(|(i, raw_transaction)| {
+                raw_transaction
+                    // FIXME: fix underlying chain of `.sign` so that it doesn't
+                    //        consume the key pair unnecessarily 
+                    .sign(genesis_key_pair.clone())
+                    .map(GenesisTransaction)
+                    .wrap_err_with(|| eyre!("Failed to sign transaction #{i}"))
```

Comment:
Depends really on personal perception =)
For me zero-based indexing seems more natural in this context. Perhaps it would be clearer to say

> Failed to sign transaction _at index_ 0